### DISCUSSION TITLE
Exit status -1 on finding zero tenants. Why?

### DISCUSSION THREAD

**Discussion by ChristiaanBilthof** [spatie/laravel-multitenancy#447] [General]
I have a question regarding the following piece of code in the trait ("Spatie\Multitenancy\Commands\Concerns\TenantAware").

```
if ($tenantQuery->count() === 0) {
      $this->error('No tenant(s) found.');

      return -1;
}
```

This code returns a -1 exit code (which results in a 255 exit-code in Bash) meaning this will fail when running a pipeline or sequence of scripts (using script concatination with '&&').

This means that when I'm deploying to a new setup, my pipeline will always fail because there are no tenants, which makes sense because my new setup does not contain any tenants yet.

So I was wondering, could this be removed (or moved into a config setting perhaps to let the user decide how they wish to handle this), or changed in favor of just a 'log' and not an error exit-code?

Maybe I've missed why this is important but I only see this as unhelpful for now.

Awesome package btw! Thanks, Chris

**Comment by masterix21** [comment]
Hi @ChristiaanBilthof, feel free to create a PR. Thanks

**Reply by ChristiaanBilthof** [reply]
Hey Luca,

Thanks for your response. Sure, will do. Just to make sure, should I just remove the if-statement? In my opinion, I do not see the relevancy of throwing an error or logging anything here; the user has a million others ways of figuring out whether they have tenants or not.

Let me know what you prefer and I'll get to it.

Best,

**Reply by ChristiaanBilthof** [reply]
@masterix21 What are your thoughts?
